daphne/src/messages/mod.rs

```diff
@@ -169,12 +175,17 @@ impl Decode for ReportMetadata {
             time: Time::decode(bytes)?,
             extensions: decode_u16_items(&(), bytes)?,
         };
-        // Check for duplicate extensions.
+        // Check for duplicate extensions and unknown extensions.
         let mut seen: HashSet<u16> = HashSet::new();
         for extension in &metadata.extensions {
             if !seen.insert(extension.type_code()) {
                 return Err(CodecError::UnexpectedValue);
             }
+            if matches!(extension, Extension::Unhandled { .. }) {
+                // TODO(bhalleycf)  Draft03 Early Input Share Validation says that unrecognized
+                // extensions are an error.  Might ignoring them be better?
```

Comment:
This was settled in https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/issues/369. (I agree, FWIW, but it was decided to abort.)